src/v/kafka/protocol/api_versions.h

```diff
@@ -72,4 +72,29 @@ inline bool operator==(
            && a.max_version == b.max_version;
 }
 
+inline bool operator==(
+  const kafka::supported_feature_key& a,
+  const kafka::supported_feature_key& b) {
+    return a.name == b.name && a.min_version == b.min_version
+           && a.max_version == b.max_version;
+}
+
+inline bool operator==(
+  const kafka::finalized_feature_key& a,
+  const kafka::finalized_feature_key& b) {
+    return a.name == b.name && a.max_version_level == b.max_version_level
+           && a.min_version_level == b.min_version_level;
+}
+
+inline bool operator==(
+  const kafka::api_versions_response& a,
+  const kafka::api_versions_response& b) {
+    return a.data.error_code == b.data.error_code
+           && a.data.throttle_time_ms == b.data.throttle_time_ms
+           && a.data.finalized_features_epoch == b.data.finalized_features_epoch
+           && a.data.api_keys == b.data.api_keys
+           && a.data.supported_features == b.data.supported_features
+           && a.data.finalized_features == b.data.finalized_features;
+}
+
```

Comment:
We discussed seeing if this can be replaced with default / compiler-generated operator==

Reply:
@dotnwat opinions on the drawbacks on this? i.e. adding a bunch of compiler generated functions to generated code that will only be used in one test

Reply:
My comment here is a nit: it doesn't need to hold up the PR but I was curious on best practices here for equality operators.

Reply:
Going to abandon this idea as not every type can have `operator==` defined for it. Thinking about produce_request which holds a `kafka::produce_request_record_data` instance in it.